solr/core/src/java/org/apache/solr/search/QueryUtils.java

```diff
@@ -215,4 +221,38 @@ public static Query combineQueryAndFilter(Query scoreQuery, Query filterQuery) {
       }
     }
   }
+
+  /**
+   * Parse the filter queries in Solr request
+   *
+   * @param req Solr request
+   * @param fixNegativeQueries if true, negative queries are rewritten by adding a MatchAllDocs
+   *     query clause
+   * @return and array of Query. If the request does not contain filter queries, returns an empty
+   *     list.
+   * @throws SyntaxError if an error occurs during parsing
+   */
+  public static List<Query> parseFilterQueries(SolrQueryRequest req, boolean fixNegativeQueries)
+      throws SyntaxError {
+
+    String[] filterQueriesStr = req.getParams().getParams(CommonParams.FQ);
+
+    if (filterQueriesStr != null) {
+      List<Query> filters = new ArrayList<>(filterQueriesStr.length);
+      for (String fq : filterQueriesStr) {
+        if (fq != null && fq.trim().length() != 0) {
+          QParser fqp = QParser.getParser(fq, req);
+          fqp.setIsFilter(true);
```

Comment:
Wondering if this could also be under a flag since not all existing callers currently do the `setIsFilter` call.
```suggestion
          if (setIsFilter) {
            fqp.setIsFilter(true);
          }
```

Reply:
I think that adding this kind of parameter might make the method more confused and would discourage future usage.  
setIsFilter(true) is needed for optimizing queries that do not need scoring but are not necessary. In the case of KnnQParser, it is needed also for understanding if it is necessary to parse recursively the filter queries (and avoid infinite recursion).
I double-checked and scoring is not involved in any scenario involved in this refactoring.

Reply:
Yes, @cpoerschke , I have done some investigations and it seems that the callers that were not setting it, should have done that, so it shouldn't be parametric